### PR TITLE
Create meeting step 1 frontend 

### PR DESCRIPTION
**Pull request by vivianho18**
Front end code and testing for the first step in the create meeting sequence.

### CODE REVIEW

**Comment by vivianho18** [comment]
Screen Shot 2021-01-19 at 2 11 28 pm

File: GoMeet/src/main/webapp/create-meeting-step1.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <br> 
+    <form id="meeting-host-form">
+      <label for="meeting-host">Enter your email: </label>
+      <input type="text" name="meeting-host" id='meeting-host'>
+    </form>
+    <br>
+    <p>Please enter guests' emails to invite them to the meeting</p>
+    <ul id="guest-list"></ul>
+    <button onclick="addGuest(document)">Add another guest</button>
+    </div>
```
Comment (ghost) [inline GoMeet/src/main/webapp/create-meeting-step1.html]: Looks like you left this in by accident?

File: GoMeet/src/main/webapp/test/spec/specCreateMeetingStep1.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  beforeAll(function() {
+    addGuest(document);
+  });
+
+  afterAll(function() {
+    removeGuest('guest-1', document); 
+  });
+
+  it ("Should add one li element to the html page", function() {
+    var listSize = document.getElementById('guest-list').childNodes.length;
```
Comment (theresenieva) [inline GoMeet/src/main/webapp/test/spec/specCreateMeetingStep1.js]: nit: Google JavaScript style guide specifies to 'Declare all local variables with either const or let. The var keyword must not be used.'

**Review by theresenieva — COMMENTED** [review]
nit: for JavaScript files, from the Google JavaScript guide: 'File names must be all lowercase and may include underscores (_) or dashes (-), but no additional punctuation.' I've used "-" in my files because we followed the same format for the HTML files too :)

File: GoMeet/src/main/webapp/js/create-meeting-step1.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  newTextInput.name = 'guest-' + guestNumber
+
+  let newInputLabel = doc.createElement('label');
+  newInputLabel.htmlFor = 'guest-' + guestNumber;
+  newInputLabel.innerText = 'Guest: ';
+
+  // delete button associated with this guest
+  let removeGuestButton = doc.createElement('button');
+  removeGuestButton.innerHTML = 'X';
+  removeGuestButton.addEventListener('click', function() {removeGuest(listItem.id, doc);});
```
Comment (ghost) [inline GoMeet/src/main/webapp/js/create-meeting-step1.js]: Have you heard of arrow function experssions? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
It's a more compact way of writing this.

File: GoMeet/src/main/webapp/js/create-meeting-step1.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  newTextInput.type = 'text';
+  newTextInput.name = 'guest-' + guestNumber
+
+  let newInputLabel = doc.createElement('label');
+  newInputLabel.htmlFor = 'guest-' + guestNumber;
+  newInputLabel.innerText = 'Guest: ';
+
+  // delete button associated with this guest
+  let removeGuestButton = doc.createElement('button');
+  removeGuestButton.innerHTML = 'X';
```
Comment (ghost) [inline GoMeet/src/main/webapp/js/create-meeting-step1.js]: Prefer using innerText here as this is just the text of the element.

File: GoMeet/src/main/webapp/js/create-meeting-step1.js (hunk 3 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/js/create-meeting-step1.js]: Would it be better to use the arrow function expression to call my removeGuest() function? 
i.e. removeGuestButton.addEventListener('click', ()=>(removeGuest(listItem.id, doc)));

Or should I use the arrow function to completely replace the removeGuest() function
i.e. removeGuestButton.addEventListener('click', ()=>(doc.getElementById(listItem.id).remove()));

Comment (ghost) [reply GoMeet/src/main/webapp/js/create-meeting-step1.js]: Either is probably fine although the former is probably easier to read.

File: GoMeet/src/main/webapp/js/create-meeting-step1.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+let guestsAdded = 0; // Number of guest email inputs on the page 
+
+/**
+ * Add an input element to the guest list for user to add email 
+ * @param {*} doc the document object to be manipulated 
+ */
+function addGuest(doc) {
```
Comment (ghost) [inline GoMeet/src/main/webapp/js/create-meeting-step1.js:7]: I think it's fine if you just use the global `document` value

File: GoMeet/src/main/webapp/create-meeting-step1.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+  <!-- Create Meeting Step 1: Set meeting name and add guests -->
+  <head>
+    <title>GoMeet Create Meeting Step 1</title>
+    <link rel="stylesheet" href="style.css">
+    <script src="js/create-meeting-step1.js"></script>
+  </head>
+  <body onload="addGuest(document);">
+    <h1 style="text-align:center;">GoMeet</h1>
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/create-meeting-step1.html]: nit: style should be in the style sheet :https://google.github.io/styleguide/htmlcssguide.html#Separation_of_Concerns.

You could consider adding a '.center' class to assign to these h1's and other elements you want to center.

File: GoMeet/src/main/webapp/index.html [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>GoMeet</title>
+    <link rel="stylesheet" href="style.css">
+  </head>
+  <body>
+    <h1 style="text-align:center;">GoMeet</h1>
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/index.html]: nit: style should be in style sheet, see other comment.

File: GoMeet/src/main/webapp/index.html [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    <a href="create-meeting-step1.html"><p style="text-align:center;">Create a New Meeting</p></a>
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/index.html]: nit: same style comment as before.

File: GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    expect(numGuests).toBe(1); 
+
+    // Remove the first guest 
+    guestList[0].childNodes[2].click();
+    numGuests = guestList.length; 
+    expect(guestOne).toBeNull; 
+    expect(guestTwo).toBeNull; 
+    expect(numGuests).toBe(0); 
+  });
+});
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js]: nit: newline at EOF

File: GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    let children = document.getElementById('guest-1').childNodes; 
+    let elementType = children[1].tagName; 
+    expect(elementType).toBe("INPUT");
+  });
+
+  it ("The third child node should be a button element", function() {
+    let children = document.getElementById('guest-1').childNodes; 
+    let elementType = children[2].tagName; 
+    expect(elementType).toBe("BUTTON");
+  });
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js]: Are these all seperate tests? If so, I think they could be combined as one, and you could add an additional test to add multiple guests.

**Review by BipBopBoop — COMMENTED** [review]
Generally looks good, just some nits and a recommendation to the tests.

File: GoMeet/src/main/webapp/create-meeting-step1.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    <h2>Create Meeting</h2>
+    <h3>Step 1: Set meeting name and add guests</h3>
+    <form id="meeting-name-form">
+      <label for="meeting-name">Enter meeting name: </label>
+      <input type="text" name="meeting-name" id='meeting-name'>
+    </form>
+    <br> 
+    <form id="meeting-host-form">
+      <label for="meeting-host">Enter your email: </label>
+      <input type="text" name="meeting-host" id='meeting-host'>
```
Comment (anna-giang) [inline GoMeet/src/main/webapp/create-meeting-step1.html]: You can consider using an `type="email"` input here - that input type will do simple checks to verify that the input was in the format of an email (i.e. contains '@', no spaces, no special characters, etc).

File: GoMeet/src/main/webapp/js/create-meeting-step1.js (hunk 5 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/js/create-meeting-step1.js:7]: I needed to dependency inject the document object in order to write my unit tests

File: GoMeet/src/main/webapp/test/spec/specCreateMeetingStep1.js (hunk 2 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/test/spec/specCreateMeetingStep1.js]: Fixed :)

File: GoMeet/src/main/webapp/create-meeting-step1.html (hunk 1 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/create-meeting-step1.html]: Oops!

File: GoMeet/src/main/webapp/js/create-meeting-step1.js (hunk 3 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/js/create-meeting-step1.js]: Went with the former

File: GoMeet/src/main/webapp/js/create-meeting-step1.js (hunk 4 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/js/create-meeting-step1.js]: Fixed :)

File: GoMeet/src/main/webapp/create-meeting-step1.html (hunk 6 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/create-meeting-step1.html]: Done :)

File: GoMeet/src/main/webapp/create-meeting-step1.html (hunk 11 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/create-meeting-step1.html]: Done :)

File: GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js (hunk 10 above)
Comment (vivianho18) [reply GoMeet/src/main/webapp/test/spec/spec-create-meeting-step1.js]: Combined and added the extra test for adding more than one guest :)

File: GoMeet/src/main/webapp/js/create-meeting-step1.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  listItem.appendChild(removeGuestButton);
+
+  let guestList = doc.getElementById('guest-list');
+  guestList.appendChild(listItem);
+}
+
+ /**
+  * Remove the guest email from the list and update the id's of the remaining guests
+  * @param {*} guest the id of the 'li' element to be removed
+  * @param {*} doc the document object to be manipulated
```
Comment (BipBopBoop) [inline GoMeet/src/main/webapp/js/create-meeting-step1.js]: nit: here and earlier. Using * as the param type is not ideal, can you specify what type the parameters should be?

**Review by BipBopBoop — APPROVED** [review]
looks good, thanks for making the changes. %nit.

**Pull request merged**